Source/ARTWebSocketTransport.m

```diff
@@ -114,7 +114,14 @@ - (void)connectForcingNewToken:(BOOL)forceNewToken {
 
                 if (error) {
                     [selfStrong.logger error:@"R:%p WS:%p ARTWebSocketTransport: token auth failed with %@", _delegate, self, error.description];
-                    [selfStrong.delegate realtimeTransportFailed:selfStrong withError:[[ARTRealtimeTransportError alloc] initWithError:error type:ARTRealtimeTransportErrorTypeAuth url:self.websocketURL]];
+                    if (error.code == 40102 /*incompatible credentials*/) {
+                        // RSA15c
+                        [selfStrong.delegate realtimeTransportFailed:selfStrong withError:[[ARTRealtimeTransportError alloc] initWithError:error type:ARTRealtimeTransportErrorTypeAuth url:self.websocketURL]];
+                    }
+                    else {
+                        // RSA4b
+                        [selfStrong.delegate realtimeTransportDisconnected:selfStrong withError:[[ARTRealtimeTransportError alloc] initWithError:error type:ARTRealtimeTransportErrorTypeAuth url:self.websocketURL]];
```

Comment:
@mattheworiordan I'm not sure about this change I made.

Regarding [one of your comments](https://github.com/ably/ably-ios/pull/518#discussion_r85763133), I'm confused with [`RSA15c`](http://docs.ably.io/client-lib-development-guide/features/#RSA15c), it states that the Realtime should transition the connection state to FAILED which is incoherent with what you said.